Scripts/utils/validate_network.py

```diff
@@ -0,0 +1,70 @@
+import utils.log as log
+import parameters.assignment as param
+
+
+def validate(network, fares=None):
+    """Validate EMME network in terms of HELMET compatibility.
+
+    Check that:
+    - all auto links have volume-delay functions defined
+    - all tram links have speed defined
+    - all transit lines have headways defined
+    - a majority of nodes has transit fare zone defined (optional)
+
+    Parameters
+    ----------
+    network : inro.emme.network.Network
+        Network to be validated
+    fares : assignment.datatypes.transit_fare.TransitFareZoneSpecification
+            Transit fare zone specification (optional)
+    """
+    if fares is not None:
+        fare_zones = fares.transit_fare_zones
+        log.debug("Zonedata has fare zones {}".format(', '.join(fare_zones)))
+        transit_zones = set()
+        nr_transit_zone_nodes = 0
+        nr_nodes = 0
+        # check that fare zones exist in network
+        for node in network.nodes():
+            nr_nodes += 1
+            if node.label in fare_zones:
+                nr_transit_zone_nodes += 1
+            transit_zones.add(node.label)
+        log.debug("Network has fare zones {}".format(', '.join(transit_zones)))
+        if fare_zones > transit_zones:
+            log.warn(
+                "All zones in transit costs do not exist in node labels.")
```

Comment:
Change warning into
- "Some zones in transit costs do not exist in node labels." or
- "Not all zones in transit costs exist in node labels."